cmake/GzPython.cmake

```diff
@@ -24,3 +24,101 @@ find_package(Python3 ${GZ_PYTHON_VERSION} QUIET)
 if(Python3_EXECUTABLE AND NOT PYTHON_EXECUTABLE)
   set(PYTHON_EXECUTABLE ${Python3_EXECUTABLE})
 endif()
+
+# Macro Definitions #
+
+#################################################
+# Insert options required to control the include of pybind11
+#
+# Usage:
+#
+#     gz_add_pybind_project_settings()
+#
+# Variables defined by this macro:
+#
+#  SKIP_PYBIND11                                Skip generating of Python bindings if OFF
+#  USE_SYSTEM_PATHS_FOR_PYTHON_INSTALLATION     Install modules in default system path if ON
+#  USE_DIST_PACKAGES_FOR_PYTHON                 Use dist-package over site-package for module installation if ON
+#
+macro(gz_add_pybind_project_settings)
```

Comment:
apologies for not reviewing this sooner

I actually deprecated `GzPython` in https://github.com/gazebosim/gz-cmake/pull/431; I think these new macros should be in a new file like `GzPybind.cmake`